lib/commanded/event_store/adapters/spear/mapper.ex

```diff
@@ -83,10 +83,14 @@ defmodule Commanded.EventStore.Adapters.Spear.Mapper do
     %RecordedEvent{data: snapshot} = event
 
     data =
-      snapshot.source_type
-      |> String.to_existing_atom()
-      |> struct(snapshot.data)
-      |> JsonDecoder.decode()
+      if is_struct(snapshot.data) do
```

Comment:
could you use an extra function clause + `is_struct` guard instead of the if statement?